src/main/java/org/folio/search/service/reindex/jdbc/CallNumberRepository.java

```diff
@@ -0,0 +1,184 @@
+    INSERT INTO %s (
+        call_number_id,
+        item_id,
+        instance_id,
+        tenant_id,
+        location_id
+    ) VALUES (?, ?::uuid, ?::uuid, ?, ?::uuid);
+    """;
+
+  protected CallNumberRepository(JdbcTemplate jdbcTemplate,
+                                 JsonConverter jsonConverter,
+                                 FolioExecutionContext context,
+                                 ReindexConfigurationProperties reindexConfig) {
+    super(jdbcTemplate, jsonConverter, context, reindexConfig);
+  }
+
+  @Override
+  public void deleteByInstanceIds(List<String> instanceIds) {
+    var sql = DELETE_QUERY.formatted(getFullTableName(context, INSTANCE_CALL_NUMBER_TABLE),
+      getParamPlaceholderForUuid(instanceIds.size()));
+    jdbcTemplate.update(sql, instanceIds.toArray());
+  }
+
+  @Override
+  public void saveAll(ChildResourceEntityBatch entityBatch) {
+    saveResourceEntities(entityBatch);
+    saveRelationshipEntities(entityBatch);
+  }
+
+  @Override
+  public ReindexEntityType entityType() {
+    return ReindexEntityType.CALL_NUMBER;
+  }
+
+  @Override
+  protected String entityTable() {
+    return CALL_NUMBER_TABLE;
+  }
+
+  @Override
+  protected Optional<String> subEntityTable() {
+    return Optional.of(INSTANCE_CALL_NUMBER_TABLE);
+  }
+
+  @Override
+  protected RowMapper<Map<String, Object>> rowToMapMapper() {
+    return null;
+  }
+
+  private void saveRelationshipEntities(ChildResourceEntityBatch entityBatch) {
+    var instanceCallNumberTable = getFullTableName(context, INSTANCE_CALL_NUMBER_TABLE);
+    var instanceCallNumberSql = INSERT_RELATIONS_SQL.formatted(instanceCallNumberTable);
+
+    jdbcTemplate.batchUpdate(instanceCallNumberSql, entityBatch.relationshipEntities(), BATCH_OPERATION_SIZE,
+      (statement, entity) -> {
+        statement.setString(1, getCallNumberId(entity));
+        statement.setString(2, getItemId(entity));
+        statement.setString(3, getInstanceId(entity));
+        statement.setString(4, getTenantId(entity));
+        statement.setString(5, getLocationId(entity));
+      });
+  }
+
+  private void saveResourceEntities(ChildResourceEntityBatch entityBatch) {
+    var callNumberTable = getFullTableName(context, entityTable());
+    var callNumberSql = INSERT_ENTITIES_SQL.formatted(callNumberTable);
+
+    jdbcTemplate.batchUpdate(callNumberSql, entityBatch.resourceEntities(), BATCH_OPERATION_SIZE,
+      (statement, entity) -> {
+        statement.setString(1, getId(entity));
+        statement.setString(2, getCallNumber(entity));
+        statement.setString(3, getPrefix(entity));
+        statement.setString(4, getSuffix(entity));
+        statement.setString(5, getTypeId(entity));
+        statement.setString(6, getVolume(entity));
+        statement.setString(7, getEnumeration(entity));
+        statement.setString(8, getChronology(entity));
+        statement.setString(9, getCopyNumber(entity));
+      });
```

Comment:
Maybe one-by-one processing on batch update failure, same as for other resources?